weave/Scripts/Utils/Unique.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+namespace weave.Utils;
+
+/// <summary>
+///     A utility class designed to provide unique identifiers and colors.
+/// </summary>
+public static class Unique
+{
+    private static int _counter = 1;
+
+    /// <summary>
+    ///     Some nice looking colors : )
+    /// </summary>
+    private static readonly ISet<Color> Colors = new HashSet<Color>
+    {
+        Godot.Colors.Turquoise,
+        Godot.Colors.Magenta,
+        Godot.Colors.SpringGreen,
+        Godot.Colors.OrangeRed,
+        Godot.Colors.Gold,
+        Godot.Colors.BlueViolet,
+        Godot.Colors.HotPink,
+        Godot.Colors.DarkOrange
+    };
+
+    private static readonly ISet<Color> UsedColors = new HashSet<Color>();
+
+    /// <summary>
+    ///     Generates a new unique identifier represented as a string.
+    /// </summary>
+    /// <returns>A string representation of a unique identifier.</returns>
+    public static string NewId()
```

Comment:
will not necessarily be unique 🫣🫣🫣🫣
maybe [https://learn.microsoft.com/en-us/dotnet/api/system.guid.newguid?view=net-7.0](this)

Reply:
@erikwessman I updated the PR before and this method is removed as the id is now their color